src/apps/exactly/common/exactly.rewards.contract-position-fetcher.ts

```diff
@@ -0,0 +1,86 @@
+import { Inject } from '@nestjs/common';
+import { constants } from 'ethers';
+
+import { APP_TOOLKIT, IAppToolkit } from '~app-toolkit/app-toolkit.interface';
+import { getLabelFromToken } from '~app-toolkit/helpers/presentation/image.present';
+import { MetaType } from '~position/position.interface';
+import { ContractPositionTemplatePositionFetcher } from '~position/template/contract-position.template.position-fetcher';
+import type {
+  GetDefinitionsParams,
+  GetTokenDefinitionsParams,
+  GetTokenBalancesParams,
+  GetDisplayPropsParams,
+  DefaultContractPositionDefinition,
+} from '~position/template/contract-position.template.types';
+
+import { ExactlyContractFactory, Previewer } from '../contracts';
+
+import {
+  ExactlyDefinitionsResolver,
+  type ExactlyMarketDefinition,
+  PREVIEWER_ADDRESSES,
+} from './exactly.definitions-resolver';
+import type { ExactlyMarketProps } from './exactly.token-fetcher';
+
+type ExactlyDefinition = DefaultContractPositionDefinition & { exactly: ExactlyMarketDefinition[] };
+
+export abstract class ExactlyRewardsFetcher extends ContractPositionTemplatePositionFetcher<
+  Previewer,
+  ExactlyMarketProps,
+  ExactlyDefinition
+> {
+  groupLabel = 'Rewards';
+  isExcludedFromExplore = true;
+
+  constructor(
+    @Inject(APP_TOOLKIT) protected readonly appToolkit: IAppToolkit,
+    @Inject(ExactlyContractFactory) protected readonly contractFactory: ExactlyContractFactory,
+    @Inject(ExactlyDefinitionsResolver) protected readonly definitionsResolver: ExactlyDefinitionsResolver,
+  ) {
+    super(appToolkit);
+  }
+
+  async getDefinitions({ multicall }: GetDefinitionsParams) {
+    const address = PREVIEWER_ADDRESSES[this.network];
+    if (!address) throw new Error(`missing previewer on ${this.network}`);
```

Comment:
Line 44 and 45 won't be necessary as both lines are already in `ExactlyDefinitionsResolver.getDefinitions()`

Reply:
done, thanks!